csrc/executor.cpp

```diff
@@ -925,13 +925,18 @@ at::Tensor allocateOutput(
     return ee.evaluate(out_tv).as<at::Tensor>();
   }
 
-  if (aliased_in != nullptr) {
-    const PolymorphicValue& aliased_in_val = ee.evaluate(aliased_in);
+  if (aliased_io != nullptr) {
     NVF_ERROR(
-        aliased_in_val.is<at::Tensor>(),
+        aliased_io->isFusionInput() || aliased_io->isFusionOutput(),
+        "Otherwise, `ee.evaluate` ",
```

Comment:
nitpick, should we mention `aliased_io needs to be an input/output`. I guess the code is self-explanatory, maybe that's not necessary.

Reply:
nitpick: Is it fair to say the GPU computation is involved when we alias intermediate tensors? ee just falls to torch, so technically if we are aliasing intermediate, that intermediate could have been just a reshaped input and technically the assert message here isn't valid.

Note I understand why we don't want to support that yet, but let's fix the error message.

Reply:
Done.